diff --git a/node_modules/next/dist/server/base-server.js b/node_modules/next/dist/server/base-server.js
index 2aed864..e1ddc49 100644
--- a/node_modules/next/dist/server/base-server.js
+++ b/node_modules/next/dist/server/base-server.js
@@ -429,6 +429,13 @@ class Server {
         _log.error(err);
     }
     async handleRequest(req, res, parsedUrl) {
+        {
+           const start = performance.now()
+           res.originalResponse.on('close', () => {
+               const responseTime = performance.now() - start
+               console.log(req.originalRequest.url, res.statusCode, responseTime)
+           })
+        }
         await this.prepare();
         const method = req.method.toUpperCase();
         const rsc = isRSCRequestCheck(req) ? "RSC " : "";
