diff --git a/node_modules/pg/lib/native/client.js b/node_modules/pg/lib/native/client.js
index 88bf52c..e50cf70 100644
--- a/node_modules/pg/lib/native/client.js
+++ b/node_modules/pg/lib/native/client.js
@@ -1,12 +1,9 @@
 'use strict'
 
-// eslint-disable-next-line
-var Native
-try {
-  // Wrap this `require()` in a try-catch to avoid upstream bundlers from complaining that this might not be available since it is an optional import
-  Native = require('pg-native')
-} catch (e) {
-  throw e
+class Native {
+    constructor() {
+        throw new Error('pg-native is not installed')
+    }
 }
 var TypeOverrides = require('../type-overrides')
 var EventEmitter = require('events').EventEmitter
